units/src/amount.rs

```diff
@@ -1039,6 +1042,29 @@ impl Amount {
     /// Returns [`None`] if overflow occurred.
     pub fn checked_div(self, rhs: u64) -> Option<Amount> { self.0.checked_div(rhs).map(Amount) }
 
+    /// Checked weight division.
+    ///
+    /// Be aware that integer division loses the remainder if no exact division
+    /// can be made.  This method rounds up ensuring the transaction fee-rate is
+    /// sufficient.  If you wish to round-down, use the unchecked version instead.
+    ///
+    /// [`None`] is returned if an overflow occurred.
+    #[cfg(feature = "alloc")]
+    pub fn checked_div_by_weight(self, rhs: Weight) -> Option<FeeRate> {
+        let sats = self.0.checked_mul(1000)?;
+        let wu = rhs.to_wu();
+
+        let fee_rate = sats.checked_div(wu)?;
+
+        let result = if sats.checked_rem(wu)? != 0 {
+            fee_rate.checked_add(1)?
```

Comment:
In 3e73807c70c85be2d684846cb7fb83633a263323:

Please just do this the normal way, by adding `wu - 1` before dividing by `wu`. This is much faster, easier to read, and less code.

Reply:
Thanks, I updated it.  I had actually tried this way originally, trying to follow how `checked_weight_by_mul` works and add 999 then divide by 1000 but I kept running into incorrect results.  And this way conceptually made sense to me.  Anyway, your suggestions of add wu -1 and then divide by wu also gives the correct results for my tests and if this is standard for the project that works for me.